client/src/modules/stock/entry/entry.js

```diff
@@ -553,9 +554,50 @@ function StockEntryController(
 
     movement.lots = Stock.processLotsFromStore(vm.stockForm.store.data, vm.movement.entity.uuid);
 
-    // fix the unit price, if necessary
+    const exchangeRate = Exchange.getCurrentRate(vm.currencyId);
+
+    // If there are shipping and handling charges, distribute them to the individual lot articles
+    if (vm.originalPO[0].shipping_handling) {
+
+      // Compute the cost of the original order (in PO currency), not including shipping+handling
+      const costPO = vm.originalPO.reduce((accum, lot) => accum + lot.unit_price * lot.quantity, 0.0);
+
+      // // ??? TEMPORARY DIAGNOSTIC CODE
```

Comment:
It looks like you need to repass through this js file and just clean up the comments.

Reply:
I will remove these commented-out diagnostic items once the PR has been reviewed for correct operation.